op-deployer/pkg/deployer/pipeline/pre_state.go

```diff
@@ -0,0 +1,112 @@
+package pipeline
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/proofs/prestate"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/state"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+)
+
+// different from inspect.GenesisAndRollup()
+// - breaks the import cycle
+// - doesn't really on globalState.AppliedIntent
+// - Though pretty anti-DRY
+func GenesisAndRollup(globalState *state.State, intermediateIntent *state.Intent, chainID common.Hash) (*core.Genesis, *rollup.Config, error) {
```

Comment:
what's up with this? why can't we refactor such that this lib gets shared?

Reply:
Yeah I pushed this for the sake of giving the view of the first look of the PR.

Rest for the refactor, I had a pretty strong preference in my mind for designating apply.go to its own package making it comply with semantics of organizing subcmds into their respective packages like we do for other subcommands.

That would produce a pretty massive line diff for the sake of this PR so didn't approach such a change in the first attempt.

That being said, let me bring this function under a "common" package of sorts and make both `inspect` and `pre_state` make use of it to DRY.